OutOfSchool/OutOfSchool.WebApi/Controllers/ProviderController.cs

```diff
@@ -0,0 +1,125 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using OutOfSchool.Services.Models;
+using OutOfSchool.WebApi.Models;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OutOfSchool.WebApi.Services;
+
+namespace OutOfSchool.WebApi.Controllers
+{
+    [ApiController]
+    [Route("[controller]/[action]")]
+    [Authorize(AuthenticationSchemes = "Bearer")]
+    public class ProviderController : ControllerBase
+    {
+        private readonly ILogger<ProviderController> logger;
+        private readonly IProviderService providerService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProviderController"/> class.
+        /// </summary>
+        /// <param name="providerService">Service for Provider model.</param>
+        public ProviderController(ILogger<ProviderController> logger, IProviderService providerService)
+        {
+            this.logger = logger;
+            this.providerService = providerService;
+        }
+
+
+        /// <summary>
+        /// Get all providers from the database.
+        /// </summary>
+        /// <returns>List of all providers.</returns>
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Provider>>> GetProviders()
+        {
+            return Ok(await providerService.GetAll().ConfigureAwait(false));
+        }
+
+        /// <summary>
+        /// Get provider by it's key.
+        /// </summary>
+        /// <param name="id">The key in the database.</param>
+        /// <returns>Provider element with some id.</returns>
+        [HttpGet("{id}")]
+        public async Task<ActionResult<ProviderDTO>> GetProviderById(long id)
+        {
+            if (id == 0)
+            { 
+                return BadRequest("Id cannot be 0.");
+            }
+
+            return Ok(await providerService.GetById(id).ConfigureAwait(false));
+        }
+
+        /// <summary>
+        /// Method for creating new provider.
```

Comment:
Could be just 'Create new provider.'

Reply:
How about just CreateProvider ? Other controllers have the same naming.